var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -310,15 +323,15 @@ def determine_variants(cls, exes, version_str):
         languages, compilers = set(), {}
         for exe in exes:
             basename = os.path.basename(exe)
-            if 'gcc' in basename:
-                languages.add('c')
-                compilers['c'] = exe
-            elif 'g++' in basename:
+            if 'g++' in basename:
```

Comment:
(question) Does moving the `g++` check first fix an issue?

Reply:
Not an issue I already encountered, but I assume it's more defensive to check `gcc` last since it is the name of both the C compiler and of the compiler collection suite. That's basically the same issue for which we check the basename instead of the full path - the full path is very likely to contain the substring `gcc` in it.